### ISSUE TITLE
Pylance cannot infer the type in with statement

### ISSUE THREAD

**Issue by ArchGuyWu** [microsoft/pyright#7100]
## Environment data

-   Language Server version: 2023.12.1
-   OS and version: windows 10, 19045
-   Python version (& distribution if applicable, e.g. Anaconda): 3.11.7

## Code Snippet

```python
import ftd2xx
with ftd2xx.open() as dev:
```

## Repro Steps

1. install ftd2xx module
2. write the code

## Expected behavior

Pylance can infer the type of "dev" .

## Actual behavior

Pylance cannot infer the type of "dev" .

## Logs

```
```

**Comment by erictraut** [comment]
Pyright is working correctly here. This is an issue in the ftd2xx library. It declares the `FTD2XX` class as follows:

```python
class FTD2XX(AbstractContextManager):
    """Class for communicating with an FTDI device"""
    ...
```

The base class `AbstractContextManager` is a generic class that accepts a single type argument, but no type argument is provided here. According to the Python typing spec, type checkers should assume `Any` in this case, which is what pyright is doing. The result is that the type returned by the call to `__enter__` is `Any`, which means there's no type information available for the context manager value.

Please report the issue to the maintainers of this library.
